eslint-plugin-prettier.js

```diff
@@ -239,4 +234,30 @@ const eslintPluginPrettier = {
   },
 };
 
+// Assign configs after initial plugin configuration, as flat configs
+// require referencing the original plugin that contains the rules.
+eslintPluginPrettier.configs = {
+  recommended: {
+    extends: 'prettier',
+    plugins: ['prettier'],
+    rules: {
+      'prettier/prettier': 'error',
+      'arrow-body-style': 'off',
+      'prefer-arrow-callback': 'off',
+    },
+  },
+  ['recommended-flat']: {
+    // In the flat config, don't extend from the eslint-config-prettier ruleset.
+    // The consumer should deal with extending from eslint-config-prettier themselves.
+    plugins: {
+      prettier: eslintPluginPrettier,
```

Comment:
would it be a circular ref?

Reply:
Hmm, you're actually right! I'll make a comment about this below.